x-pack/plugins/snapshot_restore/public/application/components/repository_form/type_settings/common/max_snapshots.tsx

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+import { FormattedMessage } from '@kbn/i18n/react';
+import { EuiDescribedFormGroup, EuiFieldText, EuiFormRow, EuiTitle, EuiCode } from '@elastic/eui';
+import { RepositorySettingsValidation } from '../../../../services/validation';
+
+interface Props {
+  isInvalid: boolean;
+  defaultValue: string;
+  updateSettings: (e: React.ChangeEvent<HTMLInputElement>, settingsName: string) => void;
+  error: RepositorySettingsValidation['maxSnapshotBytesPerSec'];
+}
+
+export const MaxSnapshotsField: React.FunctionComponent<Props> = ({
+  isInvalid,
+  error,
+  defaultValue,
+  updateSettings,
+}) => {
+  return (
+    <EuiDescribedFormGroup
+      title={
+        <EuiTitle size="s">
+          <h3>
+            <FormattedMessage
+              id="xpack.snapshotRestore.repositoryForm.commonFields.maxSnapshotBytesTitle"
+              defaultMessage="Max snapshot bytes per second"
+            />
+          </h3>
+        </EuiTitle>
+      }
+      description={
+        <FormattedMessage
+          id="xpack.snapshotRestore.repositoryForm.commonFields.maxSnapshotBytesDescription"
+          defaultMessage="Maximum rate for creating snapshots for each node."
+        />
+      }
+      fullWidth
+    >
+      <EuiFormRow
+        label={
+          <FormattedMessage
+            id="xpack.snapshotRestore.repositoryForm.commonFields.maxSnapshotBytesLabel"
+            defaultMessage="Max snapshot bytes per second"
+          />
+        }
+        fullWidth
+        isInvalid={isInvalid}
+        error={error}
+        helpText={
+          <FormattedMessage
+            id="xpack.snapshotRestore.repositoryForm.commonFields.maxSnapshotBytesHelpText"
+            defaultMessage="Accepts byte size units, such as {example1}, {example2}, {example3}, or {example4}. Defaults to {defaultSize} per second."
+            values={{
+              example1: <EuiCode>1g</EuiCode>,
+              example2: <EuiCode>10mb</EuiCode>,
+              example3: <EuiCode>5k</EuiCode>,
+              example4: <EuiCode>1024B</EuiCode>,
+              defaultSize: <EuiCode>40mb</EuiCode>,
```

Comment:
I wonder if it wouldn't be better to have a constant object exported with the default sizes. We would then have a single source of truth to update them (and a single file to look at). At the same time there are only 3 values... WDYT?

Reply:
I considered this as well. I think I'm going to leave as-is for now. The `max_snapshots` setting has a specific default value, but the other two settings default to unlimited/null.